### PR TITLE
Follow best practices when handling primitive data types

### PR DESCRIPTION
**Pull request by dmitry741**
SDL T191, Follow best practices when handling primitive data types.

### CODE REVIEW

**Comment by dmitry741** [comment]
make test

File: algorithms/kernel/dbscan/dbscan_utils.h [hunk 1]
```diff
@@ -178,7 +179,8 @@ class Vector
     {
         if (_size >= _capacity)
         {
-            DAAL_CHECK_STATUS_VAR(grow());
+           services::Status status = grow();
```
Comment (ntfshard) [inline algorithms/kernel/dbscan/dbscan_utils.h]: minor: code align

**Comment by dmitry741** [comment]
make test

File: include/services/error_indexes.h [hunk 2]
```diff
@@ -414,7 +414,15 @@ enum ErrorID
 
     UnknownError = -1000000,                                            /*!< Unknown error */
     NoErrorMessageFound = -1000001,                                     /*!< No error message found */
-    ErrorMethodNotImplemented = -1000002                                /*!< Method is not implemented in the present library version  */
+    ErrorMethodNotImplemented = -1000002,                               /*!< Method is not implemented in the present library version  */
+
+    // new items for SDL T191 task: Follow best practices when handling primitive data types
```
Comment (ShvetsKS) [inline include/services/error_indexes.h]: please delete

File: service/kernel/error_handling.cpp [hunk 3]
```diff
@@ -950,6 +950,14 @@ void ErrorMessageCollection::parseResourceFile()
 
     add(NoErrorMessageFound, "NoErrorMessageFound");
     add(ErrorMethodNotImplemented, "Method is not implemented");
+
+    // new items for SDL T191 task: Follow best practices when handling primitive data types
```
Comment (ShvetsKS) [inline service/kernel/error_handling.cpp]: delete

**Comment by ShvetsKS** [comment]
fix clear code
src/algorithms/kernel/adaboost/adaboost_quality_metric_set_types.cpp
src/algorithms/kernel/adaboost/inner/adaboost_quality_metric_set_types_v1.cpp

**Comment by dmitry741** [comment]
make test

**Comment by ShvetsKS** [comment]
make test

File: algorithms/kernel/kmeans/kmeans_init_impl.i [hunk 4]
```diff
@@ -76,13 +77,16 @@ Status init( size_t p, size_t n, size_t nRowsTotal, size_t nClusters, algorithmF
...
+            DAAL_ASSERT(nRowsTotal <= services::internal::MaxVal<int>::get())
             DAAL_CHECK_STATUS(s, (UniformKernelDefault<int, cpu>::compute(i, (int)nRowsTotal, engine, 1, indices + i)));
+            DAAL_ASSERT(indices[i] >= 0)
             size_t c = (size_t)indices[i];
             int value = indices[i];
             for(size_t j = i; j > 0; j--)
             {
                 if(value == indices[j-1])
                 {
+                    DAAL_ASSERT(j-1 >= 0)
```
Comment (ShvetsKS) [inline algorithms/kernel/kmeans/kmeans_init_impl.i]: type of 'j' is size_t already, seems this check is not required

File: algorithms/kernel/linear_regression/linear_regression_quality_metric.cpp [hunk 5]
```diff
@@ -91,6 +92,7 @@ algorithms::ResultPtr ResultCollection::getResult(QualityMetricId id) const
  */
 algorithms::InputPtr InputDataCollection::getInput(QualityMetricId id) const
 {
+    DAAL_ASSERT(id >= 0)
```
Comment (ShvetsKS) [inline algorithms/kernel/linear_regression/linear_regression_quality_metric.cpp]: QualityMetricId is enum with values from 0 to 1.
extra check

File: algorithms/kernel/multiclassclassifier/multiclassclassifier_quality_metric_set_types.cpp [hunk 6]
```diff
@@ -56,6 +57,7 @@ classifier::quality_metric::multiclass_confusion_matrix::ResultPtr ResultCollect
  */
 classifier::quality_metric::multiclass_confusion_matrix::InputPtr InputDataCollection::getInput(QualityMetricId id) const
 {
+    DAAL_ASSERT(id >= 0)
```
Comment (ShvetsKS) [inline algorithms/kernel/multiclassclassifier/multiclassclassifier_quality_metric_set_types.cpp]: QualityMetricId is enum with value 0.
extra check

File: service/kernel/error_handling.cpp [hunk 7]
```diff
@@ -950,6 +950,13 @@ void ErrorMessageCollection::parseResourceFile()
 
     add(NoErrorMessageFound, "NoErrorMessageFound");
     add(ErrorMethodNotImplemented, "Method is not implemented");
+
+    add(ErrorIncorrectOffset, "Incorrect offset");
+    add(ErrorAdagradIncorrectMaxNumberOfIterations, "Incorrect maximum number of iterations value in adagrad solver parameter");
+    add(ErrorLbfgsIncorrectMaxNumberOfIterations, "Incorrect maximum number of iterations value in lbfgs parameter");
+    add(ErrorSgdIncorrectMaxNumberOfIterations, "Incorrect maximum number of iterations value in sgd parameter");
```
Comment (ShvetsKS) [inline service/kernel/error_handling.cpp]: is it agreed name for specific algorithm? why don`t we use the same error for all of them?

Comment (dmitry741) [reply service/kernel/error_handling.cpp]: These name are approved by Averbukh M.

**Comment by dmitry741** [comment]
make test

**Pull request merged**